### DISCUSSION TITLE
Variables in _quarto.yml file - folder path

### DISCUSSION THREAD

**Discussion by Juuso1** [quarto-dev/quarto-cli#5725] [Q&A]
### Description

Hi,

How can parametrize my quarto.yml file folder names in quarto?

For example :
cover-image: "{{< var general.folder >}}"

BR
Juuso

**Comment by mcanouil** [comment]
I don't understand the question.
Just do exactly as you wrote and as stated in the documentation, see <https://quarto.org/docs/authoring/variables.html#var>.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/598d0c36-0a68-4630-80aa-6dc1134455ce">

**Reply by Juuso1** [reply]
Let me explain more. If I have code like below. It gives error.

**_variable.yml file:**
general:
  folder: "themes/images/cover.png"

**Without variable it's ok**
cover-image: themes/images/cover.png

**This gives error**
cover-image:  "{{< var general.folder >}}"

**Reply by mcanouil** [reply]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by cscheid** [comment]
The issue here is that shortcodes are not resolved in the way you expect. (This is 100% the fault of our documentation; we need to specify and document precisely how shortcodes work.)

Loosely speaking, shortcodes are good when you want to resolve what's ultimately going to be markdown content (something that ends up in the page). Here, you're attempting to use shortcodes such that the result is additional metadata. That doesn't work in quarto right now.

However, there are other ways to achieve what you want. In this case, you could try using [project profiles](https://quarto.org/docs/projects/profiles.html) such that different renderings of the project can use different metadata. Since you're attempting to control a project-wide setting, I believe this would work in your case.

**Reply by Juuso1** [reply]
Thank you for your answer. I try to parametrize my project so that I have only one variable file.
